src/com/seafile/seadroid2/ui/activity/BrowserActivity.java

```diff
@@ -302,6 +300,76 @@ public void onPageScrolled(int arg0, float arg1, int arg2) {
         Intent monitorIntent = new Intent(this, FileMonitorService.class);
         startService(monitorIntent);
 
+        fetchServerInfo();
+    }
+
+    private void fetchServerInfo() {
+        boolean proEdition = isServerProEdition();
+        if (proEdition)
+            return;
+        else {
+            // hide Activity tab and search menu
+            adapter.hideActivityTab();
+            indicator.notifyDataSetChanged();
+            adapter.notifyDataSetChanged();
+            // hide search menu
+            if (menuSearch != null)
+                menuSearch.setVisible(false);
+        }
+
+        if (!Utils.isNetworkOn())
+            return;
+
+        ConcurrentAsyncTask.execute(new FetchServerInfoTask());
+    }
+
+    class FetchServerInfoTask extends AsyncTask<Void, Void, ServerInfo> {
+
+        @Override
+        protected ServerInfo doInBackground(Void... params) {
+            try {
+                return dataManager.getServerInfo();
+            } catch (IOException e) {
+                e.printStackTrace();
+            } catch (SeafException e) {
+                e.printStackTrace();
+            } catch (JSONException e) {
+                e.printStackTrace();
+            }
+            return null;
+        }
+
+        @Override
+        protected void onPostExecute(ServerInfo serverInfo) {
+            //super.onPostExecute(aVoid);
+            if (serverInfo == null)
+                return;
+
+            if (serverInfo.isProEdition()) {
+                // hide Activity tab and search menu
+                adapter.unHideActivityTab();
+                indicator.notifyDataSetChanged();
+                adapter.notifyDataSetChanged();
+                // hide search menu
+                if (menuSearch != null)
+                    menuSearch.setVisible(true);
+            }
+
+            saveServerProEdition(account.getServer(), serverInfo.isProEdition());
+        }
+    }
+
+    private static Map<String, Boolean> serverInfoMap = Maps.newHashMap();
```

Comment:
We can use a `ServerInfo` object directly, no need to covert it to a `Map` object.